server/src/services/websocket-server.ts

```
@@ -0,0 +1,461 @@
+                // Not yet authenticated ⇒ safe to purge immediately
+                cleanupSession(sessionId);
+            }
+        };
+
+        ws.on('close', topLevelCloseHandler);
+
+        ws.on('error', (err) => {
+            logError(`[Server] WebSocket error: ${err.message}`);
+            const existing = getSessions().get(sessionId);
+            if (existing && existing.authenticated) {
+                scheduleOrphanCleanup(existing);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)